python/gtsam/examples/Pose2ISAM2Example.py

```diff
@@ -0,0 +1,155 @@
+                gtsam.Pose2(2.1, 2.1, -math.pi/2),
+                gtsam.Pose2(1.9, -0.2, 0.2)]
+
+    graph.push_back(gtsam.PriorFactorPose2(1, gtsam.Pose2(0, 0, 0), PRIOR_NOISE))
+    initial_estimate.insert(1, gtsam.Pose2(0.5, 0.0, 0.2))
+
+    def determine_loop_closure(odom: np.ndarray, current_estimate: gtsam.Values,
+        xy_tol=0.6, theta_tol=0.3) -> int:
+        """Simple brute force approach which iterates through previous states
+        and checks for loop closure.
+
+        Args:
+            odom: Vector representing noisy odometry (x, y, theta) measurement in the body frame.
+            current_estimate: The current estimates computed by iSAM2.
+            xy_tol: Optional argument for the x-y measurement tolerance.
+            theta_tol: Optional argument for the theta measurement tolerance.
+        Returns:
+            k: The key of the state which is helping add the loop closure constraint.
+                If loop closure is not found, then None is returned.
+        """
+        if current_estimate:
+            prev_est = current_estimate.atPose2(i+1)
+            rotated_odom = prev_est.rotation().matrix() @ odom[:2]
+            curr_xy = np.array([prev_est.x() + rotated_odom[0],
+                                prev_est.y() + rotated_odom[1]])
+            curr_theta = prev_est.theta() + odom[2]
+            for k in range(1, i+1):
+                pose_xy = np.array([current_estimate.atPose2(k).x(),
+                                    current_estimate.atPose2(k).y()])
+                pose_theta = current_estimate.atPose2(k).theta()
+                if (abs(pose_xy - curr_xy) <= xy_tol).all() and \
+                    (abs(pose_theta - curr_theta) <= theta_tol):
+                        return k
+
+    current_estimate = None
+    for i in range(len(odom_arr)):
+        # The "ground truth" change between poses
+        odom_x, odom_y, odom_theta = odom_arr[i]
+        # Odometry measurement that is received by the robot and corrupted by gaussian noise
+        odom = mult_gauss(odom_arr[i], ODOMETRY_NOISE.covariance())
+        # Determine if there is loop closure based on the odometry measurement and the previous estimate of the state
+        loop = determine_loop_closure(odom, current_estimate)
+        if loop:
+            graph.push_back(gtsam.BetweenFactorPose2(i + 1, loop, gtsam.Pose2(odom_x, odom_y, odom_theta), ODOMETRY_NOISE))
+        else:
+            graph.push_back(gtsam.BetweenFactorPose2(i + 1, i + 2, gtsam.Pose2(odom_x, odom_y, odom_theta), ODOMETRY_NOISE))
+            initial_estimate.insert(i + 2, pose_est[i])
+        # Incremental update to iSAM2's internal Baye's tree, which only optimizes upon the added variables
+        # as well as any affected variables
+        isam.update(graph, initial_estimate)
+        # Another iSAM2 update can be performed for additional optimization
+        isam.update()
+        current_estimate = isam.calculateEstimate()
+        print("*"*50)
```

Comment:
Combine prints if possible using "\n" if needed

Reply:
Maybe combine printing and plotting in Pose2SLAM_ISAM2_plot, which, since it's an internal function, you can just call `report_on_progress`